src/setup_payload/ManualSetupPayloadGenerator.cpp

```diff
@@ -31,17 +31,60 @@
 
 namespace chip {
 
-static uint32_t shortPayloadRepresentation(const SetupPayload & payload)
+static uint32_t chunk1PayloadRepresentation(const SetupPayload & payload)
 {
-    constexpr int discriminatorOffset = kCustomFlowRequiredFieldLengthInBits;
-    constexpr int pinCodeOffset       = discriminatorOffset + kManualSetupDiscriminatorFieldLengthInBits;
-    uint32_t result                   = payload.requiresCustomFlow ? 1 : 0;
+    /* <1 digit> Represents:
+     *     - <bits 1..0> Discriminator <bits 11.10>
+     *     - <bit 2> VID/PID present flag
+     */
 
-    static_assert(kManualSetupDiscriminatorFieldBitMask <= UINT32_MAX >> discriminatorOffset, "Discriminator won't fit");
-    result |= static_cast<uint32_t>((payload.discriminator & kManualSetupDiscriminatorFieldBitMask) << discriminatorOffset);
+    constexpr int discriminatorShift     = (kPayloadDiscriminatorFieldLengthInBits - kManualSetupDiscriminatorFieldLengthInBits +
+                                        kManualSetupChunk1DiscriminatorMsbitsLength);
```

Comment:
It doesn't matter since `kManualSetupChunk1DiscriminatorMsbitsLength` == `kManualSetupChunk2DiscriminatorLsbitsLength`, but I guess the formula should contain `kManualSetupChunk2DiscriminatorLsbitsLength` instead of `kManualSetupChunk1DiscriminatorMsbitsLength`, right? Or just do `discriminatorShift = kPayloadDiscriminatorFieldLengthInBits - kManualSetupChunk1DiscriminatorMsbitsLength`.

Reply:
Good catch. I like the simpler equation. Changed